src/Orleans.Runtime/Core/InsideRuntimeClient.cs

```diff
@@ -430,6 +431,7 @@ public void SniffIncomingMessage(Message message)
 
                     if (message.Direction != Message.Directions.OneWay)
                     {
+                        TransactionContext.Clear();
```

Comment:
Same here... at first thought at least seems that it should clear before returning, regardless of direction... I might be wrong, as I didn't follow it very thoroughly

Reply:
ah, nevermind, I see now that this whole block is wrapped with a try/finally where you clear in the finally statement

---
In reply to: [154475381](https://github.com/dotnet/orleans/pull/3730#discussion_r154475381) [](ancestors = 154475381)